Assets/Scripts/Fighting/AttackCalculator.cs

```diff
@@ -2,22 +2,25 @@
 using System.Collections.Generic;
 using UnityEngine;
 
-public class AttackCalculator : MonoBehaviour
+namespace Fighting
 {
-    private GameObject _attachedGameobjekt;
-
-    private void Start()
-    {
-        _attachedGameobjekt = transform.parent.gameObject;
-    }
-    public void CalculateDamage(float damage)
+    public class AttackCalculator : MonoBehaviour
     {
-        PlayerProperties NPCProps = _attachedGameobjekt.GetComponent<PlayerProperties>();
-        NPCProps.SetHealth(Mathf.Clamp(NPCProps.health - damage, 0, NPCProps.maxHealth));
-    }
+        public GameObject attachedGameobjekt { get; set; }
```

Comment:
It can be used at runtime. It should be avoided if not necessary though.
It is well and truly ok to just load the NPCProps at startup from the current gameobject or its parent/children.

Reply:
(and rename to actual component)

Reply:
Yes, it would be ok - however it is way cleaner to just reference the props directly - so no `GetComponent` call is necessary, and the references are not hard coded, but set in the inspector.

Reply:
But using the inspector is more error prone. With this it's guaranteed to work when there is that type of component.
We should avoid unnecessary references in the inspector as they are likely to be done wrong.

Reply:
That was my take here too. I don't wont that much references in the editor, they are messy and often lead to bugs and mistakes...